### PR TITLE
CI: interface tests

### PR DESCRIPTION
**Pull request by FreyJo**
Contributions:

DONE
- [x] fix appveyor
- [x] add `test_ocp` octave
- [x] cmake install: creates shared libraries for acados, HPIPM, BLASFEO in same directory as make does, i.e. `<acados_root>/lib`
- [x] travis: compile and run the C examples
- [x] travis: add octave test for integrators

TODOs4future
- [ ] test matlab_mex interface from Matlab (maybe even different versions)
- [ ] test for templating interface - after https://github.com/acados/acados/pull/410 is merged
- [ ] make something run on appveyor: currently it only compiles

### CODE REVIEW

File: interfaces/acados_matlab/ocp_set_ext_fun.m [hunk 1]
```diff
@@ -384,7 +385,9 @@
 			cflags_tmp = [cflags_tmp, ' -DN0=', num2str(phase_start{ii})];
 			cflags_tmp = [cflags_tmp, ' -DN1=', num2str(phase_end{ii})];
 			setenv('CFLAGS', cflags_tmp);
-			mex(acados_include, acados_interfaces_include, external_include, blasfeo_include, acados_lib_path, acados_matlab_lib_path, model_lib_path, '-lacados', '-lhpipm', '-lblasfeo', ['-l', model_name], mex_files{1});
+			mex(acados_include, acados_interfaces_include, external_include, blasfeo_include,...
+				hpipm_include, acados_lib_path, acados_matlab_lib_path, model_lib_path, '-lacados',...
+				 '-lhpipm', '-lblasfeo', ['-l', model_name], mex_files{1});
 		else
 			mex(mex_flags, 'CFLAGS=$CFLAGS -std=c99 -fopenmp', ['-DSETTER=', setter{ii}], ['-DSET_FIELD=', set_fields{ii}], ['-DMEX_FIELD=', mex_fields{ii}], ['-DFUN_NAME=', fun_names{ii}], ['-DPHASE=', num2str(phase{ii})], ['-DN0=', num2str(phase_start{ii})], ['-DN1=', num2str(phase_end{ii})], acados_include, acados_interfaces_include, external_include, blasfeo_include, acados_lib_path, acados_matlab_lib_path, model_lib_path, '-lacados', '-lhpipm', '-lblasfeo', ['-l', model_name], mex_files{1}); % TODO
```
Comment (giaf) [inline interfaces/acados_matlab/ocp_set_ext_fun.m]: the fix should be done to this line too

Comment (FreyJo) [reply interfaces/acados_matlab/ocp_set_ext_fun.m]: right, good spot!

**Comment by FreyJo** [comment]
Feedback welcome.
From my side, this can be merged.

File: interfaces/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+
+# add_test(NAME matlab_test_sim_pendulum 
+#         COMMAND "${CMAKE_COMMAND}" -E chdir ${PROJECT_SOURCE_DIR}/examples/matlab_mex/pendulum_on_cart_model ${MATLAB_EXECUTABLE} -nodisplay -nodesktop -r  "run('./run_tests.m');")
+
+add_test(NAME octave_test_sim 
+        COMMAND "${CMAKE_COMMAND}" -E chdir ${PROJECT_SOURCE_DIR}/examples/matlab_mex/test
+        octave --no-gui --no-window-system ./run_tests_sim.m)
```
Comment (giaf) [inline interfaces/CMakeLists.txt]: does `octave-cli` do the same?

File: ci/linux/install_octave.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+#
+#/bin/bash
+
+sudo apt-add-repository ppa:octave/stable -y;
```
Comment (giaf) [inline ci/linux/install_octave.sh]: What is this doing? Isn't octave packages already available on any "recent" ubuntu version?

Is this giving access to a more recent version of octave or something similar?

File: interfaces/CMakeLists.txt (hunk 2 above)
Comment (FreyJo) [reply interfaces/CMakeLists.txt]: On my machine, it does not make a difference if I write 
`octave` or `octave-cli`
Might depend on the installation.

File: ci/linux/install_octave.sh (hunk 3 above)
Comment (FreyJo) [reply ci/linux/install_octave.sh]: I struggled a bit to find a good octave version.
Without this, I only got the trusty version, which is 3.x.
I ended up using this inspired by the CasADi colleagues.
https://github.com/casadi/casadi/blob/6b0de13ac78cb59b10e7a7a11d8a96d8fecbf951/.travis.yml#L332
It uses Octave 4.2.2, which is close to what we have and should support imo.
Supporting multiple Octave and Matlab versions would be desirable but is not so easy.

**Comment by giaf** [comment]
Ok it looks good, merging!

**Pull request merged**